products/views.py

```diff
@@ -0,0 +1,230 @@
+            openapi.Parameter("page", openapi.IN_QUERY, type=openapi.TYPE_INTEGER),
+        ],
+        responses={200: ProductReadSerializer(many=True)},
+    )
+    def get(self, request, *args, **kwargs):
+        return super().get(request, *args, **kwargs)
+
+
+class CurrentUserProductRetrieveView(generics.RetrieveAPIView):
+    serializer_class = ProductReadSerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get_queryset(self):
+        queryset = Product.objects.filter(creator_id=self.request.user.pk).order_by(
+            "-created_at"
+        )
+
+        return queryset
+
+    def retrieve(self, request, *args, **kwargs):
+        return super().retrieve(request, *args, **kwargs)
+
+
+class ProductCreateView(generics.CreateAPIView):
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = ProductWriteSerializer
+
+    def get_queryset(self):
+        queryset = Product.objects.filter(creator_id=self.request.user.pk).order_by(
+            "-created_at"
+        )
+
+        return queryset
+
+
+class ProductDeleteView(generics.DestroyAPIView):
+    queryset = Product.objects.all()
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = ProductWriteSerializer
+
+    def get_queryset(self):
+        queryset = Product.objects.filter(creator_id=self.request.user.pk).order_by(
+            "-created_at"
+        )
+
+        return queryset
+
+    def delete(self, request, *args, **kwargs):
+        # Get the product by its ID
+        product = self.get_object()
+
+        if not product:
+            raise exceptions.NotFound("Product not found")
+
+        # Delete the product
+        product.delete()
+
+        return response.Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class ProductBidCreateView(generics.CreateAPIView):
+    serializer_class = BidWriteSerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    def create(self, request, product_id, *args, **kwargs):
+        try:
+            product = Product.objects.get(pk=product_id)
+        except Product.DoesNotExist:
+            return response.Response(
+                {"error": "Product not found"}, status=status.HTTP_404_NOT_FOUND
+            )
+
+        if product.creator.pk == self.request.user.pk:
+            return response.Response(
+                {"error": "You cannot bid on your own product"},
+                status=status.HTTP_403_FORBIDDEN,
+            )
+
+        bid_amount = request.data.get("bid_amount")
+
+        if not bid_amount:
+            return response.Response(
+                {"error": "bid_amount is required"}, status=status.HTTP_400_BAD_REQUEST
+            )
+
+        try:
+            bid_amount = int(bid_amount)
+        except ValueError:
+            return response.Response(
+                {"error": "bid_amount must be a positive integer"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
```

Comment:
These validations should be in serializer not in view. Check how to validate data in serializer.